relay-server/src/services/store.rs

```diff
@@ -199,6 +210,10 @@ impl StoreService {
             KafkaTopic::Attachments
         } else if event_item.as_ref().map(|x| x.ty()) == Some(&ItemType::Transaction) {
             KafkaTopic::Transactions
+        } else if use_feedback_topic
```

Comment:
nit: `is_rolled_out` is called for every envelope, even if it does not contain a UserReportV2. So I would turn this condition around and evaluate `is_rolled_out` only when the event type matches.